RFC_PROXY_AGENT.md

```diff
@@ -0,0 +1,306 @@
+# Public Cluster Verification Problems
+
+## Original Problems
+
+In the original design of the Spore Protocol, the creation of a Spore with a Cluster ID required the following conditions:
+
+1. The referenced Cluster Cell must be found in the `CellDeps`.
+2. The referenced Cluster Cell must be found in the `Inputs`.
+3. The referenced Cluster Cell must be found in the `Outputs`.
+
+This verification process ensures ownership of the referenced Cluster Cell, preventing malicious activities on one's private Cluster.
+
+However, while this approach prevents unintended minting in a certain Cluster, it brings a problem for Clusters that support public minting — those using specific lock to achieve this. That is, there might be a bottleneck when the Cluster contains multiple popular Spore cells, and there is only one Cluster cell available to construct transactions.
+
+More specifically, if two person attempt to mint different Spores into a Cluster, say Cluster A, and their minting transactions are accidentally sent at the same time, then one of them will be rejected because the target Cluster was consumed and recreated by the other. The rejected one must then be reconstructed to continue the minting process.
+
+## Solution
+
+At this stage, the **owner** needs to create a special cell, here we called it **Cluster Proxy Cell**
+
+A Cluster Proxy Cell’s structure is like below:
+
+```yaml
+Cluster Proxy Cell:
+    Data: REFERENCED_CLUSTER_ID
+    Type:
+        code_hash: CLUSTER_PROXY_TYPE_HASH
+        args: <cluster_proxy_id> [<min_payment>]
```

Comment:
As we discussed, will `min_payment` be moved to `cell_data` in this RPC version or next version?

Reply:
after I complete it, I'll change this doc description about `min_payment`